### ISSUE TITLE
[a11y]: Accessibility  issue in ExpandableSearch component

### ISSUE THREAD

**Issue by kvaidyanathan-ns1** [carbon-design-system/carbon#17247]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v1.64.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

The IBM Accessibility Checker tool has identified an issue with the ExpandableSearch component in our project, which uses "@carbon/react": "1.64.0". The tool reports:
` the element with the role "button" contains descendants with the role "search," which are ignored by browsers.`
This seems to be an accessibility violation.

<img width="1720" alt="Screenshot 2024-08-22 at 17 32 32" src="https://github.com/user-attachments/assets/c61ecd8e-f0c7-4c4b-b0c6-59f85a9fbd4c">

### Reproduction/example

https://stackblitz.com/edit/github-v38m2h?file=src%2FApp.jsx

### Steps to reproduce

Go to the shared stackblitz exampe and run the  IBM Accessibility Checker tool.

### Suggested Severity

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
Hey there, thanks for reporting this! In this case I believe the violation is rooted in how you're using the components. 

It's not explicitly enforced in the types, but children passed to `HeaderGlobalAction` should be an Icon 

https://github.com/carbon-design-system/carbon/blob/d19d1f6ca116b99f392bf7f1e6aecbe496716202/packages/react/src/components/UIShell/HeaderGlobalAction.tsx#L55

`HeaderGlobalAction` wraps it's children with a Button, so when you put ExpandableSearch in it the a11y violation appears because a role of search is nested within a button.
